File: ComputerPartsEcommerce-ejb/src/java/entity/HDD.java
package entity;

import java.io.Serializable;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.validation.constraints.NotNull;

@Entity
public class HDD extends ComputerPart implements Serializable {

    private static final long serialVersionUID = 1L;

    @Column(nullable = false)
    @NotNull
    private String manufacturer;
    @Column(nullable = false)
    @NotNull
    private String type; //(RPM)
    @Column(nullable = false)
    @NotNull
    private Integer Capacity; //in GB
    @Column(nullable = false)
    @NotNull
    private String formFactor; //(3.5'')
    @Column(nullable = false)
    @NotNull
    private String Interface; //(SATA 6gb/s  SATA 3gb/s )  

    public HDD() {
    }

    public HDD(String name, Double price, Integer inventoryQuantity, String image) {
        super(name, price, inventoryQuantity, image);
    }

    public HDD(String manufacturer, String type, Integer Capacity, String formFactor, String Interface, String name, Double price, Integer inventoryQuantity, String image) {
        super(name, price, inventoryQuantity, image);
        this.manufacturer = manufacturer;
        this.type = type;
        this.Capacity = Capacity;
        this.formFactor = formFactor;
        this.Interface = Interface;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Integer getCapacity() {
        return Capacity;
    }

    public void setCapacity(Integer Capacity) {
        this.Capacity = Capacity;
    }

    public String getFormFactor() {
        return formFactor;
    }

    public void setFormFactor(String formFactor) {
        this.formFactor = formFactor;
    }

    public String getInterface() {
        return Interface;
    }

    public void setInterface(String Interface) {
        this.Interface = Interface;
    }

    public String getManufacturer() {
        return manufacturer;
    }

    public void setManufacturer(String manufacturer) {
        this.manufacturer = manufacturer;
    }
    
    
    
}

File: ComsmanCustomerClientAngular/src/app/hdd.ts
import { ComputerPart } from './computer-part'

export class HDD extends ComputerPart {

    manufacturer: string;
    type: string;
    capacity: number;
    formFactor: string;
    interfaceForm: string;

    // constructor(productId?: number, name?: string, price?: number, inventoryQuantity?: number, image?: string, isDisabled?: boolean,
    //     manufacturer?: string,
    //     type?: string,
    //     capacity?: number,
    //     formFactor?: string,
    //     Interface?: string) {

    //     super(productId, name, price, inventoryQuantity, image, isDisabled);
    //     this.manufacturer = manufacturer;
    //     this.type = type;
    //     this.capacity = capacity;
    //     this.formFactor = formFactor;
    //     this.interfaceForm = Interface;
    // }


}
